DepotManager.cs

```diff
@@ -90,19 +87,20 @@ public class DepotManager {
           return;
         }
 
-        List<int> neededDepots = new List<int>();
+        List<SteamManifest> neededManifests = new List<SteamManifest>();
+        long totalDownloadSize = 0;
         if (!IsFullyDownloaded(version, Package.Main)) {
-          neededDepots.AddRange(ManifestData.depots);
+          neededManifests.AddRange(ManifestData.GetManifestsForVersion(version, Package.Main));
+          totalDownloadSize += ManifestData.GetDownloadSize(version, Package.Main);
         }
         if (Settings.Default.ownsGehenna && !IsFullyDownloaded(version, Package.Gehenna)) {
-          neededDepots.Add(ManifestData.GEHENNA);
+          neededManifests.AddRange(ManifestData.GetManifestsForVersion(version, Package.Gehenna));
+          totalDownloadSize += ManifestData.GetDownloadSize(version, Package.Gehenna);

```

Comment:
Having several calls to GetDownloadSize doesn't seem cleaner. Why not preserve the old loop?

Reply:
The old way ends up duplicating code. This was one of the spots I was thinking a flags enum would help a decent bit, though I ended up not going with it cause it makes getting a folder per package awkward, bit of a remnant from that too.

In any case I'll revert it.